### PR TITLE
Dawnpm 2023 june release notes and q4 roadmap

### PR DESCRIPTION
**Pull request by dawnpm**
## PR Checklist
- [x] Test Cloud.gov Pages preview for your branch for functionality - make sure all links and buttons work, javascript works as intended, and updates display as expected
  - Preview link - release notes: https://cg-a72b65a7-73f0-4859-a1d7-dc953cf6ade8.sites.pages.cloud.gov/preview/gsa/search-gov-website/dawnpm-2023-June-release-notes/about/updates/releases/june-2023.html
  - Preview link - roadmap: https://cg-a72b65a7-73f0-4859-a1d7-dc953cf6ade8.sites.pages.cloud.gov/preview/gsa/search-gov-website/dawnpm-2023-June-release-notes/about/updates/roadmap/
- [x] Conduct a HemingwayApp review of the content for plain language. Request plain language review in #helpwanted if needed before merging this PR - fixes implemented
- [x] Conduct a Lighthouse Audit via Chrome Developer Tools. Identify any issues that must be addressed before merging, and fix - no issues
- [x] Ensure no non-public information is published in the commit messages

### CODE REVIEW

File: _releases/2023-07-18-june-2023.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+* **Results Page Redesign**: We've continued progress on updating our frontend components.
+  * We built the [Graphics Best Bets](https://search.gov/admin-center/content/best-bets.html#best-bets-graphics) component for the new results page.
+  * We added term-match highlighting to the backend data for results snippets.
+  * We made sure that long URLs and descriptions display in a user-friendly manner.
+  * We prepared backend data for some types of search results:
+    * Jobs 
+    * Federal Register documents
+    * Health topics
+* **Indexing improvements**: Our system now prioritizes customer requests for reindexing over general index maintenance.
+* **Results matching improvements**: In this phase we are looking at alphanumeric strings, such as form numbers or policy numbers. We created a Levenstein model that will identify variations on strings that are "close enough" to consider matches. For example, "DD 214" and "DD214."
```
Comment (MyNameIsMissing) [inline _releases/2023-07-18-june-2023.md:22]: pretty sure the spelling is Levenshtein ref: https://en.wikipedia.org/wiki/Levenshtein_distance

**Review by MyNameIsMissing — CHANGES_REQUESTED** [review]
one spelling item

**Comment by jimkreft** [comment]
All this content looks reasonable!

**Review by jimkreft — APPROVED** [review]
all this makes sense (it took me a second to understand the roadmap content shifting around, but now it makes sense)

**Comment by dawnpm** [comment]
Merging without poking Rob again. Requested change has been resolved.

**Pull request merged**